components/org.wso2.carbon.identity.api.server.workflow.engine/org.wso2.carbon.identity.rest.api.server.workflow.engine.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/workflow/engine/v1/core/WorkflowService.java

```diff
@@ -0,0 +1,631 @@
+    private boolean getAssociationState(String associationId, Status status){
+
+        boolean isEnable;
+        try {
+
+            if (status == null) {
+
+                isEnable = workflowManagementService.getAssociation(associationId).isEnabled();
+            } else {
+                isEnable = (status.getAction() == ActionStatus.ENABLE);
+            }
+            return isEnable;
+
+        } catch (WorkflowException e) {
+            throw handleServerError(e.getErrorCode(),e.getMessage(), "", e);
+        }
+    }
+
+    /**
+     * Partially update an association
+     *
+     * @param associationId  Association ID
+     * @param workflowAssociation  Association Details
+     * @throws WorkflowException
+     */
+    public String changeAssociation(String associationId, WorkflowAssociationPatch workflowAssociation){
+
+        try {
+            Association association = workflowManagementService.getAssociation(associationId);
+            if (association == null ) {
+                throw new WorkflowClientException("Invalid workflow association ID provided.");
+            }
+            boolean isEnable = getAssociationState(associationId, workflowAssociation.getAssociationStatus());
+            workflowManagementService.changeAssociation(associationId, workflowAssociation.getAssociationName(), workflowAssociation.getWorkflowId(), workflowAssociation.getOperationName(), workflowAssociation.getAssociationCondition(), isEnable);
+            return "Workflow association successfully updated!";
+
+        }  catch (WorkflowClientException e) {
+            throw handleClientError(e.getErrorCode(), e.getMessage(), "", e);
+        } catch (WorkflowException e) {
+            throw handleServerError(e.getErrorCode(), e.getMessage(), "", e);
+        }
+    }
+
+    private ErrorResponse.Builder getErrorBuilder(String errorCode, String errorMsg, String data) {
+
+        return new ErrorResponse.Builder().withCode(errorCode).withMessage(errorMsg)
+                .withDescription(includeData(errorMsg, data));
+    }
+
+    private static String includeData(String errorMsg, String data) {
+
+        String message = errorMsg;
+        if (data != null) {
+            message = String.format(errorMsg, data);
+        }
+        return message;
+    }
+
+    private APIError handleServerError(String errorCode, String errorMsg, String data, Exception e) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorCode, errorMsg, data).build(log,
+                e, includeData(errorMsg, data));
+
+        return new APIError(Response.Status.INTERNAL_SERVER_ERROR, errorResponse);
+
+    }
+
+    private APIError handleClientError(String errorCode, String errorMsg, String data, WorkflowClientException e) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorCode, errorMsg, data).build(log,
+                e, includeData(errorMsg, data));
+
+        return new APIError(Response.Status.BAD_REQUEST, errorResponse);
+
+    }
```

Comment:
```suggestion
    private APIError handleClientError(String errorCode, String errorMsg, String data, WorkflowClientException e) {

        ErrorResponse errorResponse = getErrorBuilder(errorCode, errorMsg, data).build(log,
                e, includeData(errorMsg, data));
        return new APIError(Response.Status.BAD_REQUEST, errorResponse);
    }
```

Remove unnecessary new lines. Apply same for other places.